index.js

```diff
@@ -237,11 +237,19 @@ RedisClient.prototype.create_stream = function () {
             this.stream.destroy();
         }
 
+        // Accept snake_case options and convert them to camelCase
+        var opts = {};
+        var keys = Object.keys(this.connection_options);
+        var key;
+        while (key = keys.pop()) {
+            opts[key.replace(/_([a-z])/g, function (g) { return g[1].toUpperCase(); })] = this.connection_options[key];
+        }
```

Comment:
In my opinion the solution is to leave ```connection_options``` as is, I mean do not ```clone``` it.
Instead of:
```
function RedisClient (options, stream) {
    // Copy the options so they are not mutated
    options = utils.clone(options);
    EventEmitter.call(this);
    var cnx_options = {};
    ...
}
```
Something like:
```
function RedisClient (options, stream) {    
    EventEmitter.call(this);
    var cnx_options = {};
    ...
    // Copy the options so they are not mutated
    options = utils.clone(options);
    ...
}
```